controllers/strong_consistency.go

```diff
@@ -30,10 +32,58 @@ func (r *SingleClusterReconciler) validateSCClusterState(policy *as.ClientPolicy
 		return err
 	}
 
-	removedNSes, err := r.removedNamespaces(allHostConns)
+	ignorableNamespaces, err := r.getIgnorableNamespaces(allHostConns)
 	if err != nil {
 		return err
 	}
 
-	return deployment.ValidateSCClusterState(r.Log, allHostConns, policy, removedNSes)
+	return deployment.ValidateSCClusterState(r.Log, allHostConns, policy, ignorableNamespaces)
+}
+
+func (r *SingleClusterReconciler) addedSCNamespaces(allHostConns []*deployment.HostConn) ([]string, error) {
+	var (
+		specSCNamespaces     = sets.NewString()
+		newAddedSCNamespaces = sets.NewString()
+	)
+
+	// Look inside only 1st rack. SC namespaces should be same across all the racks
+	rack := r.aeroCluster.Spec.RackConfig.Racks[0]
+	nsList := rack.AerospikeConfig.Value["namespaces"].([]interface{})
+
+	for _, nsConfInterface := range nsList {
+		if asdbv1.IsNSSCEnabled(nsConfInterface.(map[string]interface{})) {
+			specSCNamespaces.Insert(nsConfInterface.(map[string]interface{})["name"].(string))
+		}
+	}
+
+	nodesNamespaces, err := deployment.GetClusterNamespaces(r.Log, r.getClientPolicy(), allHostConns)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if SC namespaces are present in all node's namespaces, if not then it's a new SC namespace
+	for _, namespaces := range nodesNamespaces {
+		nodeNamespaces := sets.NewString(namespaces...)
+		newAddedSCNamespaces.Insert(specSCNamespaces.Difference(nodeNamespaces).List()...)
+	}
+
+	return newAddedSCNamespaces.List(), nil
+}
+
+func (r *SingleClusterReconciler) getIgnorableNamespaces(allHostConns []*deployment.HostConn) (
+	sets.Set[string], error) {
+	removedNSes, err := r.removedNamespaces(allHostConns)
```

Comment:
we call `deployment.GetClusterNamespaces` in both `removedNamespaces` as well as `r.addedSCNamespaces`, see if we can reduce calls here.